src/extlibs/hrvo/new_simulator.cpp

```diff
@@ -0,0 +1,163 @@
+#include "new_simulator.h"
+
+MotionPlanningSimulator::MotionPlanningSimulator(float time_step, const RobotConstants_t &robot_constants,
+                                                 const TeamColour friendly_team_colour):
+    robot_constants(robot_constants),
+    primitive_set(),
+    world(std::nullopt),
+    agents(),
+    friendly_robot_id_map(),
+    enemy_robot_id_map(),
+    friendly_team_colour(friendly_team_colour)
+{
+}
+
+void MotionPlanningSimulator::updateWorld(const World &world)
```

Comment:
Should be fully rewritten. With every new `world`, we should erase all previous agents and add them again. This should simplify the logic.